sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -843,6 +865,12 @@ private void messageReceived(final ChannelHandlerContext context, final RntbdRes
             statusCode == HttpResponseStatus.NOT_MODIFIED.code()) {
 
             final StoreResponse storeResponse = response.toStoreResponse(this.contextFuture.getNow(null));
+
+            if (this.serverErrorInjector != null
```

Comment:
Need to double think about it -> it can simulate timeout happens, but the diagnostics will be off, want to change into the same mode as connection delay

Reply:
Changed to inject delay during write